OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -62,48 +67,95 @@ public Turreted(ActorInitializer init, TurretedInfo info)
 		{
 			this.info = info;
 			TurretFacing = GetInitialTurretFacing(init, info.InitialFacing);
+
+			if (info.RandomRealign)
+				randomFacing = init.Self.World.SharedRandom.Next(0, 256);
```

Comment:
If it used angles relative to the body, yes. Unfortunately, that isn't what it does and was not what I signed up to do (yet).

When the turret code gets rewritten for the Attack refactor, I'll think about it. Right now that is out of scope.

On 2015-08-26, at 8:37 AM, Zimmermann Gyula <notifications@github.com> wrote:

> In OpenRA.Mods.Common/Traits/Turreted.cs:
> 
> > @@ -62,48 +67,95 @@ public Turreted(ActorInitializer init, TurretedInfo info)
> >  		{
> >  			this.info = info;
> >  			TurretFacing = GetInitialTurretFacing(init, info.InitialFacing);
> > +
> > +			if (info.RandomRealign)
> > +				randomFacing = init.Self.World.SharedRandom.Next(0, 256);
> Meh, I miswrote, I meant body facing instead of orientation. >_>
> 
> ―
> Reply to this email directly or view it on GitHub.
>